pilot/pkg/bootstrap/server.go

```diff
@@ -356,6 +389,43 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 	return nil
 }
 
+// createK8sServiceControllers creates all the k8s service controllers under this pilot
+func (s *Server) createK8sServiceControllers(serviceControllers *aggregate.Controller, args *PilotArgs) error {
+	kubectl := kube.NewController(s.kubeClient, args.Config.ControllerOptions)
+	serviceControllers.AddRegistry(
+		aggregate.Registry{
+			Name:             serviceregistry.ServiceRegistry(KubernetesRegistry),
+			ServiceDiscovery: kubectl,
+			ServiceAccounts:  kubectl,
+			Controller:       kubectl,
+		})
+
+	// Add clusters under the same pilot
+	if s.clusterStore != nil {
+		clusters := s.clusterStore.GetPilotClusters()
+		for _, cluster := range clusters {
+			kubeconfig := clusterregistry.GetClusterAccessConfig(cluster)
+			kubeCfgFile := strings.Join([]string{args.Config.ClusterStoreDir, kubeconfig}, "/")
+			_, client, kuberr := kube.CreateInterface(kubeCfgFile)
+			if kuberr != nil {
+				return multierror.Prefix(kuberr, fmt.Sprintf("failed to connect to Access API with accessconfig: %s", kubeCfgFile))
```

Comment:
We don't start the server if this func returns an error: any reason not to loop through and aggregate all errors rather than stopping at the first?

Reply:
no particular reason. Similar return can be found in other for loops in this file.

Reply:
Right, I meant aggregate all errors we can in this function before we bail. It's a frustrating UX to fix a problem, attempt to start the server again, and see another related failure; it's nice to get as many as we can in a single pass.

Reply:
+1